### DISCUSSION TITLE
metircs指标数据通过prometheus采集的问题

### DISCUSSION THREAD

**Discussion by CoolLiuzw** [bfenetworks/bfe#961] [Ideas（新功能的建议和讨论）]
当前期望通过  prometheus 进行 metrics 指标数据的采集，但是当前 bfe 的 metrics 的指标实现的方式为一个服务不同的模块分多个 metrics_path 获取数据。
eg:
http://xxxxx/monitor/tls_state?format=prometheus
http://xxxxx/monitor/http_state?format=prometheus

但是对于prometheus服务的配置来说，一个bfe服务相当于一个 job 。
![image](https://user-images.githubusercontent.com/44515608/148145914-af4a70d0-2a56-4661-b2d2-98f9dfb2d43e.png)
一个job，只能配置一个metrics_path，不支持多个metrics_path的方式

请问一下 bfe 实际使用中指标的采集的最佳实践是怎么样的？

**Comment by arrow2012** [comment]
这种应该 是 把多个path地址写到注册中心中，然后prometheus从注册中心中获取把？ 不应该在配置里面写死啊

**Comment by shiw13** [comment]
同问，metric是否应该集中到一个url采集，貌似Prometheus通常都是这么用的？

**Comment by mileszhang2016** [comment]
// 一个job，只能配置一个metrics_path，不支持多个metrics_path的方式
// 请问一下 bfe 实际使用中指标的采集的最佳实践是怎么样的？
@CoolLiuzw  在实践中，我们是针对不同的路径单独建立采集任务的

// metric是否应该集中到一个url采集，貌似Prometheus通常都是这么用的？
@shiw13 BFE的指标数据非常多，不同路径下数据的重要性（及需要的采集频度）也不一样。
如果都集中在一个url中采集，从性能的角度和灵活性的角度反而有问题。

采集BFE运行的指标确实是非常重要的功能。
我们计划Q1内将使用Prometheus监控BFE的模板也开源出来，供大家使用。
请大家给我们一些时间。谢谢。

**Reply by z8n24Xsolla** [reply]
请问 这个prometheus 的监控模版有开源了么？
